rye/src/cli/publish.rs

```diff
@@ -11,13 +11,13 @@ use crate::utils::CommandOutput;
 /// Builds a package for distribution.
 #[derive(Parser, Debug)]
 pub struct Args {
-    /// A directory containing the build artifacts to publish.
-    #[arg(long)]
-    dist: Option<PathBuf>,
+    /// The distribution files to upload to the repository (defaults to <workspace-root>/dist/*).
+    dist: Option<Vec<PathBuf>>,
     /// The repository url to publish to (default is https://upload.pypi.org/legacy/).
     #[arg(short, long, default_value = "https://upload.pypi.org/legacy/")]
```

Comment:
```suggestion
    #[arg(long, default_value = "https://upload.pypi.org/legacy/")]
```

Could reserve `-r` for `--repository`.